### ISSUE TITLE
Button variant overwrites prop style 

### ISSUE THREAD

**Issue by Patrick-Ullrich** [chakra-ui/chakra-ui#3625]
# 🐛 Bug report

Buttons variant overwrites its style props.

## 💥 Steps to reproduce

Set a variant on the button e.g. unstyled, then try to overwrite padding using style props.

## 💻 Link to reproduction

CodeSandbox reproduction: https://codesandbox.io/s/divine-sea-6n826?file=/src/App.tsx

## 🧐 Expected behavior

When adding a variant that sets a style, I expect to still be able to overwrite this style using style props. You will notice that padding from the variant is being applied after the prop style padding and therefore overwriting it.

## 🧭 Possible Solution

Prop styles should supersede the variant.
Haven't checked the code yet, but merging props and variant, or changing sort order should do the trick.

## 🌍 System information

| Software         | Version(s) |
| ---------------- | ---------- |
| Chakra UI        |      1.3.4    |
| Browser          |       chrome     |
| Operating System |      w10      |

## 📝 Additional information

Workaround for now:

```
<Button variant="unstyled" paddingLeft="1rem !important"></Button>
```

**Comment by TimKolberger** [comment]
I think this is fixed in the current version 1.5.0. Could you kindly check again?

**Comment by Patrick-Ullrich** [comment]
Confirmed, seems to be fixed in v.1.5.0
